backend/pgnService/game/update.ts

```diff
@@ -144,60 +138,83 @@ function getRequest(event: APIGatewayProxyEventV2): UpdateGameRequest {
  * @param request The UpdateGameRequest to process.
  * @returns A GameUpdate based on the given request.
  */
-async function getGameUpdate(
-    request: UpdateGameRequest,
-): Promise<[GameUpdate | null, GameImportHeaders | null]> {
+async function getGameUpdate(request: UpdateGameRequest): Promise<GameUpdate> {
     const update: GameUpdate = {
         updatedAt: new Date().toISOString(),
     };
 
     if (request.orientation) {
         update.orientation = request.orientation;
     }
-    if (request.unlisted !== undefined) {
-        update.unlisted = request.unlisted;
-        update.publishedAt = request.unlisted ? null : new Date().toISOString();
-        update.timelineId = request.unlisted
-            ? ''
-            : `${update.publishedAt?.split('T')[0]}_${uuidv4()}`;
+
+    if (request.unlisted) {
+        update.unlisted = true;
+        update.publishedAt = null;
+        update.timelineId = '';
+    }
+
+    if (request.publish) {
+        update.unlisted = false;
+        update.publishedAt = new Date().toISOString();
+        update.timelineId = `${update.publishedAt?.split('T')[0]}_${uuidv4()}`;
     }
 
+    let result: string | undefined = request.headers?.result;
     if (request.type) {
         const pgnText = (await getPgnTexts(request))[0];
         const game = getGame(undefined, pgnText, request.headers);
-        const headers = getImportHeaders(game);
-        if (!request.unlisted && headers) {
-            return [null, headers];
-        }
 
         update.white = game.white;
         update.black = game.black;
         update.date = game.date;
-        update.headers = game.headers;
         update.pgn = game.pgn;
+        update.headers = game.headers;
+
+        result = game.headers['Result'] ?? result;
+
+        if (isMissingData({ ...update, result }) && !request.unlisted) {
```

Comment:
This check should probably be against `update.unlisted` instead of `request.unlisted`. What if I passed both `publish: true` and `unlisted: true`? My reading of this code is that it would allow me to publish with missing data.

Reply:
Since request.unlisted was (changed now) always required, request.unlisted was more relevant than update.unlisted.  I also mistakenly thought undefined unlisted meant unlisted (i.e. games defaulted to unlisted).